src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -364,6 +364,12 @@ public ReadOnlySpan<byte> ReadSpan(int len)
             return new ReadOnlySpan<byte>(Buffer, ReadPosition, len);
         }
 
+        public ReadOnlyMemory<byte> ReadMemory(int len)
+        {
+            Debug.Assert(len <= ReadBytesLeft);
+            return new ReadOnlyMemory<byte>(Buffer, ReadPosition, len);
```

Comment:
Are you sure that this will work with sequential access?

Reply:
This method is only supposed to be called when a check has already been made that the buffer contains everything (hence the assertion). In this case we have [the check](https://github.com/npgsql/npgsql/blob/a1f6f1b7f02022310d2a1911eca3dc7b553895a1/src/Npgsql/TypeHandlers/JsonHandler.cs#L262) in JsonHandler, which branches on whether the bytes are already buffered (efficient read) or not (very inefficient read...).

Another thing that will hopefully disappear soon :)